src/syntaxtree.c

```diff
@@ -158,6 +185,74 @@ static Obj SyntaxTreeDefaultCompiler(Obj result, Expr expr)
     return result;
 }
 
+// TODO: use the following as start of a "reverse compiler"
+static Expr SyntaxTreeDefaultCoder(Obj node)
+{
+
+    Obj  typeString = ElmPRec(node, RNamName("type"));
+    UInt tnum = FIND_TNUM_FROM_STRING(CSTR_STRING(typeString));
+
+    if (tnum == LAST_EXPR_TNUM + 2) {
+        ErrorQuit("Unrecognized syntax tree node type", 0, 0);
+    }
+
+    CompilerT comp = Compilers[tnum];
+
+    if (comp.code != SyntaxTreeDefaultCoder) {
+        return comp.code(node);
+    }
+
+    // compute how many slots we need
```

Comment:
```suggestion
    // compute how big the statement/expression we are generating needs to
    // be, i.e., how many Obj-sized "slots" we need; this is usually
    // comp.arity, but we need to take into account variadic argument lists
```
Not sure if this multiline insertion will work... never tried that before...